### PR TITLE
feat(dnsservices) : Secondary zones Data source

### PR DESCRIPTION
**Pull request by sandeep645**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by sandeep645** [comment]
Acceptance tests output
image

**Comment by sandeep645** [comment]
@hkantare @kavya498 
Could you please do a code review for this PR and https://github.com/IBM-Cloud/terraform-provider-ibm/pull/3904 ?

File: ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+func DataSourceIBMPrivateDNSSecondaryZones() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMDNSSecondaryZonesRead,
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The GUID of the private DNS service instance.",
```
Comment (sheltons) [inline ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go:38]:
```suggestion
				Description: "The GUID of the DNS Services instance.",
```

File: ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+						},
+						pdnsSZEnabled: {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Enable/Disable the secondary zone.",
+						},
+						pdnsSZTransferFrom: {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The addresses of DNS servers where the secondary zone data should be transferred from.",
```
Comment (sheltons) [inline ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go:74]:
```suggestion
							Description: "The addresses of DNS servers where the secondary zone data is transferred from.",
```

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+subcategory: "DNS Services"
+layout: "ibm"
+page_title: "IBM : Secondary Zones"
+description: |-
+  Manages IBM Cloud Infrastructure private domain name service secondary zones.
+---
+
+# ibm_dns_custom_resolver_secondary_zones
+
+Provides a read-only data source for secondary zones. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax. For more information about secondary zones, refer to [list-secondary-zones](https://cloud.ibm.com/apidocs/dns-svcs#list-secondary-zones)
```
Comment (sheltons) [inline website/docs/d/dns_custom_resolver_secondary_zones.html.markdown:11]:
```suggestion
Provides a read-only data source for secondary zones. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax. For more information about secondary zones, refer to [list-secondary-zones](https://cloud.ibm.com/apidocs/dns-svcs#list-secondary-zones).
```

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	instance_id	= ibm_dns_custom_resolver.test.instance_id
+	resolver_id	= ibm_dns_custom_resolver.test.custom_resolver_id
+}
+```
+
+## Argument reference
+
+Review the argument reference that you can specify for your data source.
+
+- `instance_id` - (Required, String) The GUID of the private DNS service instance.
```
Comment (sheltons) [inline website/docs/d/dns_custom_resolver_secondary_zones.html.markdown:26]:
```suggestion
- `instance_id` - (Required, String) The GUID of the DNS Services instance.
```

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+- `resolver_id` - (Required, String) The unique identifier of a custom resolver.
+
+## Attribute reference
+
+In addition to the argument references list, you can access the following attribute references after your data source are created.
```
Comment (sheltons) [inline website/docs/d/dns_custom_resolver_secondary_zones.html.markdown:31]:
```suggestion
In addition to the argument references list, you can access the following attribute references after your data sources are created.
```

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+- `secondary_zones` (List) List of secondary zones.
+
+	Nested scheme for `secondary_zones`:
+	- `description` - (String) Descriptive text of the secondary zone.
+	- `zone` - (String) The name of the zone.
+	- `enabled` - (String) Enable/Disable the secondary zone.
+	- `transfer_from` - (List) The addresses of DNS servers where the secondary zone data should be transferred from.
```
Comment (sheltons) [inline website/docs/d/dns_custom_resolver_secondary_zones.html.markdown:39]:
```suggestion
	- `transfer_from` - (List) The addresses of DNS servers where the secondary zone data is transferred from.
```

File: ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go (hunk 2 above)
Comment (sandeep645) [reply ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go]: Updated.

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown (hunk 3 above)
Comment (sandeep645) [reply website/docs/d/dns_custom_resolver_secondary_zones.html.markdown]: Updated.

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown (hunk 5 above)
Comment (sandeep645) [reply website/docs/d/dns_custom_resolver_secondary_zones.html.markdown]: Updated.

Comment (sandeep645) [reply website/docs/d/dns_custom_resolver_secondary_zones.html.markdown]: Updated.

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown (hunk 4 above)
Comment (sandeep645) [reply website/docs/d/dns_custom_resolver_secondary_zones.html.markdown]: Updated.

File: website/docs/d/dns_custom_resolver_secondary_zones.html.markdown (hunk 6 above)
Comment (sandeep645) [reply website/docs/d/dns_custom_resolver_secondary_zones.html.markdown]: Updated.

File: ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+	sess, err := meta.(conns.ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instanceID := d.Get(pdnsInstanceID).(string)
+	resolverID := d.Get(pdnsSZResolverID).(string)
+
+	opt := sess.NewListSecondaryZonesOptions(instanceID, resolverID)
+
+	result, resp, err := sess.ListSecondaryZonesWithContext(context, opt)
```
Comment (hkantare) [inline ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go:107]: Does this handles pagination if you have more than 50 or 100 secondary zones .... Will this list all or only the first page. If only first page then you need to add pagination support

Comment (vasuvall) [reply ibm/service/dnsservices/data_source_ibm_private_dns_secondary_zones.go:107]: Maximum number of secondary zones allowed for now is 5. So this is not an issue for this release.

**Comment by hkantare** [comment]
fix the conflicts

**Comment by sandeep645** [comment]
fixed the conflicts. 
It should be good now.

**Pull request merged**